pkg/disruption/backend/sampler/remote.go

```diff
@@ -0,0 +1,501 @@
+package sampler
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	projv1 "github.com/openshift/api/project/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	projectclient "github.com/openshift/client-go/project/clientset/versioned"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	monitorserialization "github.com/openshift/origin/pkg/monitor/serialization"
+)
+
+const (
+	namespace            = "e2e-disruption-monitor"
+	serviceAccountName   = "disruption-monitor-sa"
+	varLogPath           = "/var/log"
+	disruptionDataFolder = "disruption-data"
+	disruptionTypeEnvVar = "DISRUPTION_TYPE_LABEL"
+	inClusterEventsFile  = "junit/AdditionalEvents__in_cluster_disruption.json"
+)
+
+var disruptionDataPath = fmt.Sprintf("%s/%s", varLogPath, disruptionDataFolder)
+
+func TearDownInClusterMonitors(config *rest.Config) error {
+	ctx := context.Background()
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	projectClient, err := projectclient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	deleteTestBed(ctx, projectClient, kubeClient)
+
+	nodes, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	var events monitorapi.Intervals
+	var errs []error
+	for _, node := range nodes.Items {
+		nodeEvents, err := fetchNodeInClusterEvents(ctx, &node)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		framework.Logf("in-cluster monitors: found %d events for node %s", len(events), node.Name)
+		events = append(events, nodeEvents...)
+	}
+	if len(errs) > 0 {
+		framework.Logf("found errors fetching in-cluster data: %v", errs)
+	}
+	artifactPath := filepath.Join(os.Getenv("ARTIFACT_DIR"), inClusterEventsFile)
+	return monitorserialization.EventsToFile(artifactPath, events)
+}
+
+func StartInClusterMonitors(ctx context.Context, config *rest.Config) error {
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	configClient, err := configclient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	infra, err := configClient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	internalAPI, err := url.Parse(infra.Status.APIServerURL)
+	if err != nil {
+		return err
+	}
+	apiIntHost := strings.Replace(internalAPI.Hostname(), "api.", "api-int.", 1)
+
+	projectClient, err := projectclient.NewForConfig(config)
```

Comment:
no, create a namespace with a generated name with openshift-something- prefix and the right PSA labels.